### ISSUE TITLE
Maybe it's about some errors during dataset reading

### ISSUE THREAD

**Issue by xxcj-cyk** [OuyangWenyu/torchhydro#12]
I used same 29 basins from the Changdian region across two datasets to train the LSTM model. One dataset is named FD_source, and the other contains only the 29 Changdian basins. Both datasets share the same time periods, basin IDs, and variables.
﻿
But I discovered an issue: the scaler used for the "streamflow" variable (in the dapengscaler_stat.json) differed between the two datasets. To investigate further, I conducted the following experiments:
﻿
1. Running the LSTM model with 29 basins from both datasets produced different results.
2. Running the model with 28 basins, excluding changdian_95350, resulted in consistent scalers and similar outcomes across the datasets.
﻿
Upon deeper analysis, I found that the data is stored in NC files, with basins grouped in batches of 100 basins. However, changdian_95350 is stored in a separate file, while the other basins are grouped together in the same file. This discrepancy likely caused changes in the order of watersheds when reading the data, which, in turn, affected the normalization process due to differences in statistical calculations.
﻿
This finding highlights a potential issue with data handling that could introduce errors in the model's results. Below are some of the observed results from the experiments.

![image](https://github.com/user-attachments/assets/0277fb15-8004-4aa0-9c9c-f06a75c2b9d8)
![image](https://github.com/user-attachments/assets/2385e10d-cbb1-4427-9234-86fdf049de32)
![image](https://github.com/user-attachments/assets/064cf694-9a36-40d9-b2ba-0f35466646f3)

**Comment by OuyangWenyu** [comment]
Yes, this is a bug when reading data using SelfMadeHydrodataset class in hydrodatasource package. Basin IDs should sort basins, or some may be put in the wrong indices. I will release a new version for hydrodatasource

**Comment by OuyangWenyu** [comment]
Hi @xxcj-cyk, this issue has been fixed in hydrodatasource:

- cb33c51: fix a bug for reading cachenc file — sort by basin id
- 6918b47: sort after xr concat

The basin ordering bug that caused inconsistent scaler statistics when basins span multiple NC files has been resolved. Please update hydrodatasource to v0.3.0+ to get the fix.
